draft-ietf-add-ddr.md

```diff
@@ -171,44 +178,50 @@ Encrypted Resolver name.
 
 ## Authenticated Discovery {#authenticated}
 
+If the Unencrypted Resolver's IP address is a Public IP, it is considered a
+global reference identity and is subject to cryptographic authentication.
+
 In order to be considered an authenticated Designated Resolver, the
-TLS certificate presented by the Encrypted Resolver MUST contain both the domain
-name (from the SVCB answer) and the IP address of the designating Unencrypted
+TLS certificate presented by the Encrypted Resolver MUST contain the IP address
+of the designating Unencrypted
 Resolver within the SubjectAlternativeName certificate field. The client MUST
-check the SubjectAlternativeName field for both the Unencrypted Resolver's IP
-address and the advertised name of the Designated Resolver. If the
+check the SubjectAlternativeName field for the Unencrypted Resolver's IP
+address. If the
 certificate can be validated, the client SHOULD use the discovered Designated
 Resolver for any cases in which it would have otherwise used the
-Unencrypted Resolver. If the Designated Resolver has a different IP
-address than the Unencrypted Resolver and the TLS certificate does not cover the
-Unencrypted Resolver address, the client MUST NOT use the discovered Encrypted
+Unencrypted Resolver. Otherwise, the client SHOULD NOT use the discovered Encrypted
 Resolver. Additionally, the client SHOULD suppress any further queries for
 Designated Resolvers using this Unencrypted Resolver for the length of
 time indicated by the SVCB record's Time to Live (TTL).
 
-If the Designated Resolver and the Unencrypted Resolver share an IP
-address, clients MAY choose to opportunistically use the Encrypted Resolver even
-without this certificate check ({{opportunistic}}).
-
 ## Opportunistic Discovery {#opportunistic}
 
-There are situations where authenticated discovery of encrypted DNS
-configuration over unencrypted DNS is not possible. This includes Unencrypted
-Resolvers on non-public IP addresses whose identity cannot be confirmed using
-TLS certificates.
+If the Unencrypted Resolver's IP address is a Private IP, it is considered
+a local reference identity that cannot be confirmed using TLS certificates.
+Instead, the ability to respond to queries confirms control of the address.
+The client SHOULD repeat the discovery process at least every 5 minutes (e.g. by
+limiting the SVCB response TTL) to confirm that the result is still valid, and
+MUST stop using any Encrypted Resolver that is no longer designated.
```

Comment:
Rather than asking clients to repeat the discovery process, it would seem better to ask them to limit the validity of the referral. I'm trying here to avoid idle clients (like a phone with the screen off) generating unnecessary queries and draining battery. An idle device could then choose to let the referral expire, and defer the rediscovery to the time when it becomes active again.
```suggestion

The client SHOULD limit the validity of the discovered information (e.g. by
limiting the SVCB response TTL) to no more than 5 minutes, and MUST stop
using any Encrypted Resolver that is no longer designated. {{Optimizations}}
describes how to avoid periods of unencrypted resolution that could result
from this.

```